fractional-NFT/project/token-distributor-contract/tests/functions/withdraw.rs

```diff
@@ -0,0 +1,168 @@
+use crate::utils::{
+    asset_abi_calls::mint_and_send_to_address,
+    nft_abi_calls::{approve, mint},
+    test_helpers::{defaults, setup},
+    token_distributor_abi_calls::{create, purchase, token_distribution, withdraw},
+};
+use fuels::{
+    prelude::{Address, Identity},
+    signers::Signer,
+    tx::AssetId,
+};
+
+mod success {
+
+    use super::*;
+
+    #[tokio::test]
+    async fn withdraws_tokens() {
+        let (
+            _deployer,
+            owner1,
+            owner2,
+            _token_distributor_contract,
+            fractional_nft_contract,
+            nft_contract,
+            asset_contract,
+        ) = setup().await;
+        let (reserve_price, token_price, token_supply, purchase_amount, asset_supply) =
+            defaults().await;
+
+        let owner_identity = Identity::Address(owner1.wallet.address().into());
+        let fractional_nft_identity = Identity::ContractId(fractional_nft_contract.into());
+
+        mint(1, &owner1.nft, owner_identity.clone()).await;
+        approve(Some(fractional_nft_identity.clone()), &owner1.nft, 0).await;
+        create(
+            &owner1.token_distributor,
+            asset_contract.clone(),
+            fractional_nft_contract.clone(),
+            nft_contract.clone(),
+            Some(reserve_price),
+            Some(owner_identity.clone()),
+            token_price,
+            token_supply,
+            0,
+        )
+        .await;
+        mint_and_send_to_address(
+            asset_supply,
+            &owner2.asset,
+            Address::new(*owner2.wallet.address().hash()),
+        )
+        .await;
+        purchase(
+            purchase_amount,
+            &owner2.token_distributor,
+            asset_contract.clone(),
+            fractional_nft_contract.clone(),
+            token_price,
+        )
+        .await;
+
+        let token_distribution_struct =
+            token_distribution(&owner1.token_distributor, fractional_nft_contract.clone()).await;
+        assert_eq!(
+            owner1
+                .wallet
+                .get_asset_balance(&AssetId::new(*asset_contract))
+                .await
+                .unwrap(),
+            0
+        );
+        assert_eq!(
+            token_distribution_struct.clone().unwrap().external_deposits,
+            (purchase_amount * token_price)
+        );
+
+        withdraw(&owner1.token_distributor, fractional_nft_contract.clone()).await;
+
+        let token_distribution_struct =
+            token_distribution(&owner1.token_distributor, fractional_nft_contract.clone()).await;
+        assert_eq!(
+            owner1
+                .wallet
+                .get_asset_balance(&AssetId::new(*asset_contract))
+                .await
+                .unwrap(),
+            purchase_amount * token_price
+        );
```

Comment:
This happens a lot throughout many tests. Perhaps abstract away into a call so that the files only take 1 line instead of multiple?